extensions/amp-story/0.1/audio.js

```diff
@@ -0,0 +1,429 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from '../../../src/log';
+
+/**
+ * The volume to which audio should be reduced when other audio sources are
+ * played.
+ * @const {number}
+ */
+const REDUCED_VOLUME = 0.5;
+
+/**
+ * @const {number}
+ */
+const VOLUME_CHANGE_DURATION_MS = 500;
+
+/**
+ * @const {!function(number): number}
+ */
+const VOLUME_EASING_FN = input => input;
+
+/**
+ * @const {string}
+ */
+const PLAYABLE_ID_PREFIX = 'i-amphtml-playable-';
+
+
+/**
+ * @const {string}
+ */
+const BACKGROUND_AUDIO_ELEMENT_CLASS_NAME = 'i-amphtml-story-background-audio';
+
+
+/**
+ * Adds support for the background-audio property on the specified element.
+ * @param {!Element} element The element to upgrade with support for background
+ *     audio.
+ */
+export function upgradeBackgroundAudio(element) {
+  if (element.hasAttribute('background-audio')) {
+    const audioEl = element.ownerDocument.createElement('audio');
+    const audioSrc = element.getAttribute('background-audio');
+    audioEl.setAttribute('src', audioSrc);
+    audioEl.setAttribute('preload', 'auto');
+    audioEl.setAttribute('loop', '');
+    audioEl.setAttribute('autoplay', '');
```

Comment:
is there a point in adding `autoplay` given mobile ignores it and desktop will soon ignore it too?

Reply:
They only ignore it if it's unmuted, right?  Added `muted` attribute.

Reply:
not sure what they do for "audio", autoplay for muted audio is sort of strange although you may get some preloading advantages by using autoplay attr? regardless, this looks fine to me.